### PR TITLE
Create a new path_utils library in kythe/cxx/common that doesn't depend on llvm

### PR DESCRIPTION
**Pull request by justbuchanan**
I'll need these utility functions for the proto indexer.

### CODE REVIEW

**Comment by creachadair** [comment]
May I humbly request a more descriptive commit log for this change? What's there makes sense, but doesn't give much context. A line or two about why we're pulling this out into our own library would make future code archaeologists less likely to curse our names. :)

File: kythe/cxx/common/path_utils.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+#ifndef KYTHE_CXX_COMMON_PATH_UTILS_H_
+#define KYTHE_CXX_COMMON_PATH_UTILS_H_
+
+#include <string>
+
+#include "absl/strings/string_view.h"
+
+namespace kythe {
```
[the remark this answers is not in the document]
Comment (shahms) [reply kythe/cxx/common/path_utils.h]: If C++ namespace lookup rules were sensible, yes. https://abseil.io/tips/130

In this case, I *think* `kythe` is a sufficiently unique top level namespace that putting these functions inside a nested `filesystem` namespace might be reasonable (that is both specific enough and sufficiently generic to reduce the likelihood of collision), but does conflict with the current Abseil/Google "best practices" somewhat (but those are intentionally squishy).

File: kythe/cxx/common/path_utils.cc [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      return "//";
+    default:
+      return "/";
+  }
+}
+
+}  // namespace
+
+std::string JoinPath(absl::string_view a, absl::string_view b) {
+  if (absl::EndsWith(a, "/")) {
```
Comment (shahms) [inline kythe/cxx/common/path_utils.cc]: this is just:

absl::ConsumeSuffix(&a, "/");
absl::ConsumePrefix(&b, "/");

or even just:

return absl::StrCat(absl::StripSuffix(a, "/"), "/", absl::StripPrefix(b, "/"));

File: kythe/cxx/common/path_utils.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (justbuchanan) [reply kythe/cxx/common/path_utils.h]: I'll go ahead and keep it just in the toplevel `kythe` namespace if that's ok. C++ namespace rules are a bit weird... The recommendations in that abseil tip make sense for avoiding accidental surprises.

File: kythe/cxx/common/path_utils.cc (hunk 2 above)
Comment (justbuchanan) [reply kythe/cxx/common/path_utils.cc]: Much better, thanks!

**Comment by justbuchanan** [comment]
@creachadair, sure thing - I'll add some more detail to the squash/merge commit message.

**Pull request merged**